### ISSUE TITLE
pyjnius 1.4.2 and Cython 3.0.0

### ISSUE THREAD

**Issue by Tahnan** [kivy/pyjnius#662]
Hi there!  We were poking at our dependencies, one of which is pyjnius, and discovered that--very, very likely because of the Cython 3.0 release--when we try to `pip install pyjnius==1.4.2`, we get the error:

```
Building wheels for collected packages: pyjnius
  Building wheel for pyjnius (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for pyjnius (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [34 lines of output]
      warning: [options] bootstrap class path not set in conjunction with -source 7
      warning: [options] source value 7 is obsolete and will be removed in a future release
      warning: [options] target value 7 is obsolete and will be removed in a future release
      warning: [options] To suppress warnings about obsolete options, use -Xlint:-options.
      4 warnings
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-13-arm64-cpython-311
      copying jnius_config.py -> build/lib.macosx-13-arm64-cpython-311
      copying setup_sdist.py -> build/lib.macosx-13-arm64-cpython-311
      creating build/lib.macosx-13-arm64-cpython-311/jnius
      copying jnius/__init__.py -> build/lib.macosx-13-arm64-cpython-311/jnius
      copying jnius/env.py -> build/lib.macosx-13-arm64-cpython-311/jnius
      copying jnius/signatures.py -> build/lib.macosx-13-arm64-cpython-311/jnius
      copying jnius/reflect.py -> build/lib.macosx-13-arm64-cpython-311/jnius
      creating build/lib.macosx-13-arm64-cpython-311/jnius/src
      creating build/lib.macosx-13-arm64-cpython-311/jnius/src/org
      creating build/lib.macosx-13-arm64-cpython-311/jnius/src/org/jnius
      copying jnius/src/org/jnius/NativeInvocationHandler.java -> build/lib.macosx-13-arm64-cpython-311/jnius/src/org/jnius
      copying jnius/src/org/jnius/NativeInvocationHandler.class -> build/lib.macosx-13-arm64-cpython-311/jnius/src/org/jnius
      running build_ext
      warning: jnius/config.pxi:1:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: jnius/config.pxi:3:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: jnius/jnius.pyx:102:0: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: jnius/config.pxi:1:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: jnius/config.pxi:3:0: The 'DEF' statement is deprecated and will be removed in a future Cython version. Consider using global variables, constants, and in-place literals instead. See https://github.com/cython/cython/issues/4310
      warning: jnius/jnius_jvm_dlopen.pxi:57:4: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      warning: jnius/jnius.pyx:113:0: The 'IF' statement is deprecated and will be removed in a future Cython version. Consider using runtime conditions or C macros instead. See https://github.com/cython/cython/issues/4310
      Compiling jnius/jnius.pyx because it changed.
      [1/1] Cythonizing jnius/jnius.pyx
      building 'jnius' extension
      error: unknown file type '.pxi' (from 'jnius/jni.pxi')
      [end of output]
```

(Using Python 3.11.4, pip 23.2, on a Mac--not sure what information, if any, would be helpful here.)

For our purposes, we've solved the problem by unpinning and just using pyjnius 1.5, but if 1.4.2 is a thing you're still intending to support, there may be a small fix needed for it.

**Comment by ravigumm** [comment]
We are having the same issue on s390x architecture also. With 1.5 version also and with older cython versions also.

**Comment by misl6** [comment]
We're not extending support for `1.4.x`, we're only releasing bugfixes for `1.5.x` (and working on new features of course)

@ravigumm can you please open a new issue with the build log on `s390` ?

**Comment by ravigumm** [comment]
ok. Thanks. I created https://github.com/kivy/pyjnius/issues/663 for s390x side issue

**Comment by Tahnan** [comment]
> We're not extending support for `1.4.x`, we're only releasing bugfixes for `1.5.x` (and working on new features of course)

That's not only a perfectly acceptable answer, it's pretty much the one I expected. :-)  In the meantime, we've of course solved the problem on our end by updating to 1.5.  Thanks for all the work!
